commonmark/src/main/java/org/commonmark/internal/inline/BackslashInlineParser.java

```diff
@@ -0,0 +1,35 @@
+package org.commonmark.internal.inline;
+
+import org.commonmark.internal.util.Escaping;
+import org.commonmark.node.HardLineBreak;
+import org.commonmark.node.Node;
+import org.commonmark.node.Text;
+
+import java.util.regex.Pattern;
+
+/**
+ * Parse a backslash-escaped special character, adding either the escaped  character, a hard line break
+ * (if the backslash is followed by a newline), or a literal backslash to the block's children.
+ */
+public class BackslashInlineParser implements InlineContentParser {
+
+    private static final Pattern ESCAPABLE = Pattern.compile('^' + Escaping.ESCAPABLE);
+
+    @Override
+    public ParsedInline tryParse(InlineParserState inlineParserState) {
+        Scanner scanner = inlineParserState.scanner();
+        // Backslash
+        scanner.next();
+
+        char next = scanner.peek();
+        if (next == '\n') {
+            scanner.next();
+            return ParsedInline.of(new HardLineBreak(), scanner.position());
+        } else if (ESCAPABLE.matcher(String.valueOf(next)).matches()) {
+            scanner.next();
+            return ParsedInline.of(new Text(String.valueOf(next)), scanner.position());
+        } else {
+            return ParsedInline.of(new Text("\\"), scanner.position());
+        }
```

Comment:
Just to simplify those if/else and have less indentation. 

```suggestion
        if (next == '\n') {
            scanner.next();
            return ParsedInline.of(new HardLineBreak(), scanner.position());
        }

        if (ESCAPABLE.matcher(String.valueOf(next)).matches()) {
            scanner.next();
            return ParsedInline.of(new Text(String.valueOf(next)), scanner.position());
        } 

        return ParsedInline.of(new Text("\\"), scanner.position());
```